internal/handlers/assisted_service_client.go

```diff
@@ -0,0 +1,143 @@
+package handlers
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	log "github.com/sirupsen/logrus"
+)
+
+type AssistedServiceClient struct {
+	assistedServiceScheme string
+	assistedServiceHost   string
+	client                *http.Client
+}
+
+const fileRouteFormat = "/api/assisted-install/v2/infra-envs/%s/downloads/files"
```

Comment:
Isn't it better to import the existing client? similar to the installer and the agent?

Reply:
If we published the client separately then sure, but I don't want to have assisted-service as a dependency for two endpoints.

Reply:
my pr was pushed, we can probably add the client as a sub module as well it should be easy now.
Will try to work on that. 
I will check what controller and installer are using

Reply:
Yeah I think using the client would be the correct direction long term, but since this PR is just moving these functions around rather than adding anything new I'd rather leave changing the client around for a follow-up effort.

Specifically I'm concerned that the way we push auth from the image service request through to the assisted request may be difficult to translate to the generated client. I might be wrong, but we'd need to investigate this.

Reply:
good point about auth, by default it is set on creation.